script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -352,6 +352,15 @@ class PubspecCheckCommand extends PackageLoopingCommand {
             'a topic. Add "$topicName" to the "topics" section.';
       }
     }
+    for (final String topic in topics) {
+      final RegExp expectedTopicFormat =
+          RegExp(r'^[a-z](?:[a-z0-9]*[a-z0-9])?(-[a-z0-9]+)*$');
+      if (!expectedTopicFormat.hasMatch(topic)) {
+        return 'Invalid topic value "$topic" in "topics" section. '
+            'Topics must consist of lowercase alphanumerical characters or dash (but no double dash), '
+            'starting with a-z and ending with a-z or 0-9.';
```

Comment:
We should mention the length requirements as well, and/or give a link to the pubspec docs page from my comment above.